Changes to links in sidebar, fix tooltips

javfg opened the pull request:
This PR addresses some additions and fixes to the sidebar links:

* Adds an `isEos` setting into `config.json` as discussed in https://github.com/owncloud/web/issues/6702#issuecomment-1117256159
* Adds direct link to the side bar (maybe related to https://github.com/owncloud/web/issues/6840)
* Adds an eos path to the side bar, shown if `isEos`
* Ellipsizes long links so the sidebar does not get a scrollbar. Along with https://github.com/owncloud/owncloud-design-system/pull/2137, these two should close https://github.com/owncloud/web/issues/6849.
* Normalized some sidebar table method names (some of them were `titleSomething`, all are now `labelSomething`)
* Removed the colons and made the headers of the table `nowrap`. This makes the table looks better, as there is no need to have colons in headers; and they are not extremely long

Current looks:

![image](https://user-images.githubusercontent.com/6058151/168080252-40cb6244-aa9a-47ab-ab5f-95e4c12f6e74.png)

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -131,16 +198,85 @@ export default defineComponent({
     },
 
+    isEos() {
+      return !!this.configuration.options?.eos
```
Please add the eos config options to https://owncloud.dev/clients/web/getting-started/#options (markdown file in the `docs/` directory in this repo) and ideally also with a default value in https://github.com/owncloud/web/blob/72fba4481b9d3b9592ed064b0f39d8503518df66/packages/web-runtime/src/store/config.js#L40

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+          <th scope="col" class="oc-pr-s" v-text="eosPathLabel" />
+          <td>
+            <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
+              <div class="oc-flex oc-flex-middle">
```
The wrapping div around the `<p>` tag is not necessary

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+          <th scope="col" class="oc-pr-s" v-text="directLinkLabel" />
+          <td>
+            <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
+              <div class="oc-flex oc-flex-middle">
```
The wrapping div around the `<p>` tag is not necessary

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+                />
+              </div>
+              <oc-button
+                oc-tooltip="Copy EOS path"
```
Please use a `label` prop and `this.$gettext()` here

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+              <oc-button
+                oc-tooltip="Copy EOS path"
+                appearance="raw"
+                aria-label="Copy EOS path"
```
Please use a `label` prop and `this.$gettext()` here

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+                oc-tooltip="Copy EOS path"
+                appearance="raw"
+                aria-label="Copy EOS path"
+                class="oc-files-private-link-copy-url"
```
The class looks wrong

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+                />
+              </div>
+              <oc-button
+                oc-tooltip="Copy direct link"
```
Please use a `label` prop and `this.$gettext()` here

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+              <oc-button
+                oc-tooltip="Copy direct link"
+                appearance="raw"
+                aria-label="Copy direct link"
```
Please use a `label` prop and `this.$gettext()` here

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+                oc-tooltip="Copy direct link"
+                appearance="raw"
+                aria-label="Copy direct link"
+                class="oc-files-private-link-copy-url"
```
The class looks wrong

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+                :variation="copiedDirect ? 'success' : 'passive'"
+                @click="copyDirectLinkToClipboard"
+              >
+                <span text="EOS Path" />
```
Please use a `label` prop and `this.$gettext()` here

pascalwengerter replied:
Actually already exists as `eosPathLabel`, just needs to be used

javfg replied:
I'm thinking I should remove these (this and line 101). They are not showing because `text` is not a valid attribute so they are effectively empty. If I make them have something inside, we get this:

![image](https://user-images.githubusercontent.com/6058151/168782825-92d2583f-15fd-4ae1-bac7-e8aa48d584e4.png)

I assume there is no reason to have text there, but correct me if I'm wrong (pinging @diocas).

pascalwengerter replied:
You could use `v-text` and a computed property to show text (`text` itself won't work), but agreed that you could use an icon for "visual" usage and then both aria-label and tooltip on the wrapping button for contextual/screenreader help

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -78,6 +78,70 @@
+                :variation="copiedEos ? 'success' : 'passive'"
+                @click="copyEosPathToClipboard"
+              >
+                <span text="EOS Path" />
```
Please use a `label` prop and `this.$gettext()` here

javfg replied:
Same as https://github.com/owncloud/web/pull/6959#discussion_r874611497.

pascalwengerter commented on packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:
```diff
@@ -131,16 +198,85 @@ export default defineComponent({
+      return !!this.configuration?.options?.runningOnEos
+    },
+
+    hasSharees() {
```
From here to line 279 are not necessary in this component (possibly rebase-gone-wrong?), could you double-check and delete if you agree?

javfg replied:
Correct, this was included by mistake. I'll remove it.

pascalwengerter reviewed (changes requested):
Just tried and it mostly looks good, have some code hygiene/a11y points to fix but that should be a quick one :)

javfg commented:
I've applied all the suggestions (also removed some more stuff which didn't belong, like a getter used in the unneeded code you mentioned).

Also, rebased to latest master.

Let me know if I missed something!
